Upstreaming a set of fixes from Sailfish's packaging

Tomin1 opened the pull request:
This is upstreaming first set of fixes from [Sailfish's firejail packaging](https://github.com/sailfishos/firejail/). These were developed as part of implementing firejail sandboxing in Sailfish OS. See also #3960 for discussion.

These are general fixes to firejail touching logging, private-etc, memory handing and symbolic link construction around /proc filesystem. Please see the commit messages for explanations. These were written by my colleague and previously reviewed by me or one of my other colleagues.

reinerh commented:
One of these changes causes a regression in the test suite:
https://github.com/netblue30/firejail/pull/3998/checks?check_run_id=1933635335
```
Private /etc installed in 22.23 ms
Error mount bind: fs_etc.c:194 fs_private_dir_mount: No such file or directory
Error: proc 4439 cannot sync with peer: unexpected EOF
Peer 4441 unexpectedly exited with status 1
runner@fv-az102-314:~/work/firejail/firejail/test/profiles$ TESTING ERROR 0
```
(the other errors are mostly the same)

It might be related to this line:
```
fs_private_dir_mount("/usr/etc", RUN_USR_ETC_DIR);
```
because `/usr/etc` doesnot exist in the test environment.

Tomin1 commented:
Nice catch from the test suite! I added the same check that fs_private_dir_copy() has to fs_private_dir_mount() which results in some duplicated work but I think one more stat() is not that bad of a tradeoff. Also added those asserts just in case.

netblue30 commented:
all set, thanks!
